pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -889,10 +891,11 @@ func builtAutoPassthroughFilterChains(push *model.PushContext, proxy *model.Prox
 					subsetStatPrefix = util.BuildStatPrefix(push.Mesh.OutboundClusterStatName, string(service.Hostname), subset.Name, port, &service.Attributes)
 				}
 				filterChains = append(filterChains, &filterChainOpts{
-					sniHosts:       []string{subsetClusterName},
-					match:          &listener.FilterChainMatch{ApplicationProtocols: allIstioMtlsALPNs},
-					tlsContext:     nil, // NO TLS context because this is passthrough
-					networkFilters: buildOutboundNetworkFiltersWithSingleDestination(push, proxy, subsetStatPrefix, subsetClusterName, subset.Name, port, destinationRule),
+					sniHosts:   []string{subsetClusterName},
+					match:      &listener.FilterChainMatch{ApplicationProtocols: allIstioMtlsALPNs},
+					tlsContext: nil, // NO TLS context because this is passthrough
+					networkFilters: buildOutboundNetworkFiltersWithSingleDestination(
+						push, proxy, subsetStatPrefix, subsetClusterName, subset.Name, port, destinationRule, tunnelingconfig.NilBuilder),
```

Comment:
Why ? Does the API say that a DR tunnel is not supported for Gateway ?

Reply:
I forgot to mention about this fact in the API, but tunneling can't be applied to `AUTO_PASSTHROUGH` gateway, because it ignores virtual services. The idea behind this API was that a user can perform a request to the target server from an app, then route this request to a tunnel proxy with a virtual service and then apply proper destination rule which configures istio-proxy to tunnel the downstream connection. It can't be achieved without virtual service and since `AUTO_PASSTHROUGH` gateway is not matched with routing rules defined in virtual services, I don't think that there is a sense to support it.

Maybe it might work if a request would be performed directly to the tunnel proxy. Then a destination rule with tunnel settings might be applied if is defined for a host without subsets. But I don't know if it's important to support it for now - I am not sure if it's useful for users. If you think that it's important, I can do that.

Reply:
I understand now envoy doesn't support tunneling for 'pass through' - but I think it's something we should fix. See my 'gateway improvements' proposal: it can't be matched with a virtual service since there is none, but we can still tunnel the TCP connection over CONNECT using the IP. 

But I guess it needs to be implemented in envoy first.

Reply:
My assumption to not support `AUTO_PASSTHROUGH` wasn't good. @howardjohn also expected that the tunnel settings will be applied, so I will fix it.